### ISSUE TITLE
Could NOT find LibCrypto (missing: LibCrypto_LIBRARY)

### ISSUE THREAD

**Issue by BarberEngineering** [aws/aws-iot-device-sdk-cpp-v2#91]
Just grabbed the source this morning which included the patch that says it fixed the cmake/crypto issues but it's still broken for me.

Trying to do a cross compile from a linux box to an arm (i.mx6) device.

the crypto libraries are .so and are present in the usual directories (/lib)

I've tried setting the PREFIX and INSTALL dirs to where the crypto libraries are located but still doesn't work..

any suggestions?

**Comment by JonathanHenson** [comment]
Could you provide us with org cmake and build output please?

**Comment by BarberEngineering** [comment]
i've tried changing install prefix to different things, heres one: 
cmake -DCMAKE_INSTALL_PREFIX="./install"  -DBUILD_DEPS=ON ../aws-iot-device-sdk-cpp-v2
-- Packaging currently only supported on Fedora.
-- Using optimized x86_64 assembly for PQ crypto
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find LibCrypto (missing: LibCrypto_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  aws-common-runtime/aws-crt-cpp/aws-common-runtime/s2n/cmake/modules/FindLibCrypto.cmake:61 (find_package_handle_standard_args)
  aws-common-runtime/aws-crt-cpp/aws-common-runtime/s2n/CMakeLists.txt:175 (find_package)


-- Configuring incomplete, errors occurred!

**Comment by JonathanHenson** [comment]
I don’t see CMAKE_PREFIX_PATH being set. Also could you ls your install directory for me?

**Comment by BarberEngineering** [comment]
I tried setting that to my SDK directory for the dev board ($SDKDIR) (root dir containing /bin, /boot /dev /lib, etc) and tried setting it to $SDKDIR/lib

libcrypto.so sits in $SDKDIR/lib along with libcrypt.so1, libcrypt-2.19.so, libcrypto.so.1.0.0

**Comment by JonathanHenson** [comment]
Those are shared libs, if there are no libcrypto.a files you need to specify -DBUILD_SHARED_LIBS=ON

**Comment by BarberEngineering** [comment]
Jonathan, thanks!

that did the trick. I did hit one error but looking through the Cmake file I found it was trying to use my x86 libs for s2n, then I specified S2N_NO_PQ_ASM=ON and everything compiled. Now to test :)

For those that find this later here's the command (from the aws-iot-device-sdk-cpp-v2-build directory per the readme):
`cmake -DCMAKE_INSTALL_PREFIX="./install" -DBUILD_SHARED_LIBS=ON -DBUILD_DEPS=ON -DS2N_NO_PQ_ASM=ON ../aws-iot-device-sdk-cpp-v2`
